engine/runtime/src/main/java/org/enso/interpreter/node/expression/builtin/interop/syntax/HostValueToEnsoNode.java

```diff
@@ -61,8 +66,13 @@ Text doString(String txt) {
   }
 
   @Specialization(guards = {"o != null", "nulls.isNull(o)"})
-  Object doNull(Object o, @CachedLibrary(limit = "3") InteropLibrary nulls) {
-    return EnsoContext.get(this).getBuiltins().nothing();
+  Object doNull(
```

Comment:
What would happen if we removed the whole `doNull` conversion? As I wrote at [this comment](https://github.com/enso-org/enso/issues/5672#issuecomment-1434038092) I have a feeling we could run without converting `null` into `Nothing`. Especially knowing there can be _"many flavors of Nothing"_ with different warnings.

CCing @kustosz , @Akirathan , @hubertp .

Reply:
I would leave this specialization as Radek implemented. That way, you will have conversion from, e.g., Python's `None` to `Nothing`, which has a special treatment in Enso, and also all the warnings will be retained if the object was created in Enso, passed to Python, and then back to Enso again.